components/text_recognition/browser/text_recognition_win.cc

```diff
@@ -0,0 +1,346 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/text_recognition/browser/text_recognition.h"
+
+#include <windows.foundation.collections.h>
+#include <windows.globalization.h>
+#include <windows.graphics.imaging.h>
+#include <windows.media.ocr.h>
+#include <wrl/client.h>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "base/logging.h"
+#include "base/win/core_winrt_util.h"
+#include "base/win/post_async_results.h"
+#include "base/win/scoped_hstring.h"
+#include "brave/components/l10n/common/locale_util.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "services/shape_detection/detection_utils_win.h"
+#include "services/shape_detection/public/mojom/textdetection.mojom.h"
+
+// Referred shape_detection::TextDetectionImpl.
+namespace text_recognition {
+
+namespace {
+
+using ABI::Windows::Foundation::IAsyncOperation;
+using ABI::Windows::Foundation::Collections::IVectorView;
+using ABI::Windows::Globalization::ILanguage;
+using ABI::Windows::Globalization::ILanguageFactory;
+using ABI::Windows::Globalization::Language;
+using ABI::Windows::Graphics::Imaging::ISoftwareBitmap;
+using ABI::Windows::Graphics::Imaging::ISoftwareBitmapStatics;
+using ABI::Windows::Media::Ocr::IOcrEngine;
+using ABI::Windows::Media::Ocr::IOcrEngineStatics;
+using ABI::Windows::Media::Ocr::IOcrLine;
+using ABI::Windows::Media::Ocr::IOcrResult;
+using ABI::Windows::Media::Ocr::OcrLine;
+using ABI::Windows::Media::Ocr::OcrResult;
+using base::win::GetActivationFactory;
+using base::win::ScopedHString;
+using Microsoft::WRL::ComPtr;
+
+class TextDetectionImplWin {
```

Comment:
personally I prefer move classes to separate files, it make files smaller and easier to understand(+ easier for testing), it is optional though

Reply:
btw do we actually need a class here? the logic seems like linear and not requires to keep some states between calls?

Reply:
IMO, using separated class is helpful because ocr detection is done in async way so each request's result callback should be kept for each language's request.

Reply:
Moved to separeted file.